packages/api/utils/provisioningJob.ts

```diff
@@ -0,0 +1,24 @@
+import { CloudServiceProvider } from "../models/CloudServiceProvider";
+import { ProvisioningRequestType } from "../models/ProvisioningStatus";
+
+/**
+ * Create a provisioning job that will be placed on the provisioning submit
+ * queue to then be processed by the state machine. The provisioning job is
+ * generic to fit the format of different types being provisioned (e.g.,
+ * applications, operators, etc.)
+ *
+ * @param body - The request body submitted to provision resources
+ * @param type - The type of provisioning (e.g., full portfolio, applications, etc)
+ * @param csp - The CSP to provision the resources
+ */
+export class ProvisioningJob<T> {
```

Comment:
Each `/submit` should start a separate execution of the step function, since each message submitted to the queue will be polled individually and start a separate execution. 

If we are looking to process multiple jobs into a single execution, we would probably use the `Map` or `Parallel` states to process pending parts (assuming they are all submitted together and ready). If we are trying to make one CSP API invocation, we may want to prevent submissions if there are pending parts that are not ready to be provisioned, otherwise we would have to wait and have some other logic bring the pending parts together and then proceed to the transformation. 

I was thinking only completed resources ready to be provisioned would be submitted, so I did not think of this.

Reply:
OK. Let's plan to add that capability into a separate ticket. Do you have enough context to write one up yourself?

Reply:
I think I have an okay amount of context to write start the ticket, but more is better. I will at least add a ticket and put what I think the context is and we can adjust it.

Reply:
@jeffsegal-ccpo  added ticket [AT-6896](https://ccpo.atlassian.net/browse/AT-6896) let me know if this matches up with what you were thinking.